Console/Command/CakePHPUtilShell.php

```diff
@@ -0,0 +1,330 @@
+				$this->out('The messages buffer is full, no further messaged were cached.');
+			}
+		} else if(empty($this->_exceptions)) {
+			$this->out(self::SUCCESS_MESSAGE);
+		}
+	}
+
+	/**
+	 * Does the exact same thing as parent::out(), but it checks if the $this->debug is set to 1
+	 * Outputs a single or multiple messages to stdout. If no parameters
+	 * are passed outputs just a newline.
+	 *
+	 * ### Output levels
+	 *
+	 * There are 3 built-in output level.  Shell::QUIET, Shell::NORMAL, Shell::VERBOSE.
+	 * The verbose and quiet output levels, map to the `verbose` and `quiet` output switches
+	 * present in  most shells.  Using Shell::QUIET for a message means it will always display.
+	 * While using Shell::VERBOSE means it will only display when verbose output is toggled.
+	 *
+	 * @param string|array $message  A string or a an array of strings to output
+	 * @param integer      $newlines Number of newlines to append
+	 * @param integer      $level    The message's output level, see above.
+	 *
+	 * @return integer|boolean Returns the number of bytes returned from writing to stdout.
+	 * @link http://book.cakephp.org/2.0/en/console-and-shells.html#Shell::out
+	 */
+	protected function _out($message = null, $tag = false, $newlines = 1, $level = Shell::NORMAL) {
+		if ($this->_debug === 1) {
+			if ($tag) {
+				$message = sprintf('<%1$s>%2$s</%1$s>', $tag, $message);
+			}
+			return $this->out($message, $newlines, $level);
+		}
+		# because we did not write anything to the stdout
+		return 0;
+	}
+
+	/**
+	 * Sets the protected _debug variable to a certain value.
+	 *
+	 * @param int $value The debug value you would like to have.
+	 *                   By default it will read the debug value from the Configure class
+	 */
+	protected function _setDebug($value = null) {
+		if ($value === null) {
+			if (is_null($this->_debug)) {
+				# no value has been given, so we are going to read the value of the Config
+				$this->_debug = Configure::read('debug') === 0 ? 0 : 1;
+			}
+		} else {
+			$this->_debug = $value;
+		}
+	}
+
+	/**
+	 * Add an exception to the _exceptions array.
+	 *
+	 * @param Exception $e The exception, as thrown by the Try Catch block.
+	 */
+	protected function _exception($e) {
+		$options = (defined('DEBUG_BACKTRACE_IGNORE_ARGS')) ? DEBUG_BACKTRACE_IGNORE_ARGS : null;
+		$trace = debug_backtrace($options);
+		$msg = sprintf('%d: %s (%s:%d)', $e->getCode(), $e->getMessage(), $e->getFile(), $e->getLine());
+		$this->_exceptions[] = $msg;
+		$this->out(sprintf('[!] PARSE ERROR %s', $msg));
+		$this->_out(sprintf('-- Triggered in %s on line %d', $trace[1]['class'], $trace[0]['line']));
+	}
+
+	public function testProtected($methodName, array $args) {
+		return call_user_func_array(array($this, $methodName), $args);
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function _getTrace() {
+		$options = (defined('DEBUG_BACKTRACE_IGNORE_ARGS')) ? DEBUG_BACKTRACE_IGNORE_ARGS : null;
+		$trace = debug_backtrace($options);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$trace". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://dev.expandonline.nl:9000/sonarqube/coding_rules#rule_key=php%3AS1488)